js/noms/src/list.js

```diff
@@ -59,18 +78,33 @@ export default class List<T: Value> extends Collection<IndexedSequence<any>> {
                     hashValueBytes)).then(s => List.fromSequence(s));
   }
 
+  /**
+   * Insert returns a new list where `values` have been inserted at `idx`.
+   */
   insert(idx: number, ...values: Array<T>): Promise<List<T>> {
     return this.splice(idx, 0, ...values);
   }
 
+  /**
+   * Returns a new list where a single element at index `idx` have been removed.
+   */
   remove(idx: number): Promise<List<T>> {
     return this.splice(idx, 1);
   }
 
+  /**
+   * This returns a new list where `values` have been appended to the resulting list.
+   */
   append(...values: Array<T>): Promise<List<T>> {
     return this.splice(this.length, 0, ...values);
   }
 
+  /**
+   * This calls a function for every value in the list. This function is async and returns a promise
+   * that will be fulfilled when all the callback functions have been called. If the callback
+   * function returns a promise `forEach` will continue but it will not return until all of those
+   * promises have been fulfilled.
+   */
   async forEach(cb: (v: T, i: number) => ?Promise<void>): Promise<void> {
```

Comment:
Unrelated, but this reminded me: we should make the callback type `(v: T, i: number) => ?Promise<any>` not `(v: T, i: number) => ?Promise<void>`